src/test/java/io/appium/java_client/pagefactory_tests/widget/tests/combined/CombinedWidgetTest.java

```diff
@@ -161,4 +165,33 @@ public List<PartiallyCombinedWidget> getWidgets() {
             return multipleWidgets;
         }
     }
+
+
+    /**
+     * Assert proxy class cache growth for this test class.
+     * The (@link io.appium.java_client.proxy.Helpers#CACHED_PROXY_CLASSES) should be populated during these tests.
+     * Prior to the Caching issue being resolved
+     * - the CACHED_PROXY_CLASSES would grow indefinitely, resulting in an Out Of Memory exception.
+     * - this ParameterizedTest would have the CACHED_PROXY_CLASSES grow to 266 entries.
+     */
+    private void assertProxyClassCacheGrowth() {
+        System.gc(); //Trying to force a collection for more accurate check numbers
+        int thresholdSize = 50;
+        assertThat(
+            "Proxy Class Cache threshold is " + thresholdSize,
+            getCachedProxyClassesSize(),
+            lessThan(thresholdSize)
+        );
+    }
+
+    private int getCachedProxyClassesSize() {
+        try {
+            Field cpc = Class.forName("io.appium.java_client.proxy.Helpers").getDeclaredField("CACHED_PROXY_CLASSES");
+            cpc.setAccessible(true);
+            Map<?, ?> cachedProxyClasses = (Map<?, ?>) cpc.get(null);
+            return cachedProxyClasses.size();
+        } catch (Exception e) {
```

Comment:
Consider catching specific exceptions rather than a generic one. We don't want to wrap them all